src/python/gem5/simulate/simulator.py

```diff
@@ -560,13 +564,22 @@ def get_last_exit_event_cause(self) -> str:
         """
         Returns the last exit event cause.
         """
-        return self._last_exit_event.getCause()
+        return (
+            last_exit_event.getCause()
+            if (last_exit_event := self._last_exit_event)
+            else ""
+        )
 
     def get_last_exit_event_code(self) -> int:
         """
         Returns the last exit event status code
         """
-        return self._last_exit_event.getCode()
+        # TODO: determine sane default
+        return (
+            last_exit_event.getCode()
+            if (last_exit_event := self._last_exit_event)
+            else -1
+        )
```

Comment:
Does -1 work as a sane default, or should something else be used?

Raising an exception is another option too because as we discussed, this shouldn't happen if people are doing things correctly